File: App/src/models/core/Project.ts
export const ProjectStatus = {
  Planning: 0,
  InProgress: 1,
  Completed: 2,
  OnHold: 3,
} as const;

export type ProjectStatus = typeof ProjectStatus[keyof typeof ProjectStatus];

export interface Project {
  id: string;
  createdAt: string;
  updatedAt?: string;
  teamId?: string;
  name: string;
  code?: string;
  status: ProjectStatus;
  description?: string;
  clientName?: string;
  startedAt?: string;
  endedAt?: string;
  isActive: boolean;
}

File: Web2/Workenest.ProjectService/ProjectService/ProjectService.Domain/Entities/Project/Project.cs
namespace Worknest.Domain.Entities
{
    public class Project
    {
        public Guid Id { get; set; }

        public Guid? TeamId { get; set; }

        public string Name { get; set; } = string.Empty;

        public string? Code { get; set; }

        public ProjectStatus Status { get; set; } = ProjectStatus.Planning;

        public string? Description { get; set; }

        public string? ClientName { get; set; }

        public DateOnly? StartedAt { get; set; }

        public DateOnly? EndedAt { get; set; }

        public bool IsActive { get; set; } = true;

        public DateTime CreatedAt { get; set; }

        public DateTime? ModifiedAt { get; set; }

        public Team? Team { get; set; }

        public ICollection<Employee> Members { get; set; } = [];
    }
}
